ticker_test.go

```diff
@@ -0,0 +1,109 @@
+package cticker
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockTicker struct {
+	c      chan time.Time
+	done   chan struct{}
+	period time.Duration
+	times  []time.Time
+}
+
+func (t *mockTicker) run() {
+	tick := time.NewTicker(t.period)
+	defer tick.Stop()
+
+	for _, tm := range t.times {
+		fmt.Println("run:", tm)
```

Comment:
Are Println() calls intentional? (more below)

Reply:
Yes absolutely, I left them in temporarily so it gives visability when you download and test.

We will remove these before merge.